site/src/app/Header.tsx

```diff
@@ -39,8 +39,8 @@ interface Props {
 }
 
 function Header(props: Props): JSX.Element {
-    function handleTextSearchChange(e: ChangeEvent<HTMLInputElement>) {
-        props.dispatchFilters({ type: 'textSearch', payload: e.target.value });
+    function handleSearch(type: 'appSearch' | 'ncsSearch') {
```

Comment:
This `'appSearch' | 'ncsSearch'` is already specified in `filters.ts`. So either extract it from `FilterAction['type']` or export it explicitly from `filters.ts` and use that.

```suggestion
    function handleSearch(type: FilterAction['type']) {
```